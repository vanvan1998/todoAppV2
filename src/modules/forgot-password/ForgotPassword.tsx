'use client';

import React, { useState } from 'react';
import { useAuth } from 'src/contexts/AuthContext';
import Link from 'next/link';
import { Layout } from '../../components';
import styled from 'styled-components';
import { styles } from './ForgotPassword.styles';
import { SuccessText, ErrorText, Header, PlaceholderTitle } from 'src/theme';
import { Button, Input } from 'src/components';
import { useMediaQuery } from 'src/hooks';
import { isEmpty } from 'lodash';

const Container = styled.div<{ isMobile: boolean }>`
  ${styles.container}
`;

const LinkWrapper = styled.div`
  ${styles.linkWrapper}
`;

const ErrorWrapper = styled.div<{ isMobile: boolean }>`
  ${styles.errorWrapper}
`;

const SubHeader = styled.div`
  ${styles.subHeader}
`;

export const ForgotPassword = () => {
  const [email, setEmail] = useState('');
  const { resetPassword } = useAuth();
  const [error, setError] = useState('');
  const [message, setMessage] = useState('');
  const [loading, setLoading] = useState(false);
  const { isMobile } = useMediaQuery();

  const handleSubmit = async (e: any) => {
    e.preventDefault();

    try {
      setMessage('');
      setError('');
      setLoading(true);
      await resetPassword(email);
      setMessage('Check your inbox for further instructions');
    } catch {
      setError('Failed to reset password');
    }

    setLoading(false);
  };

  return (
    <Layout>
      <Container isMobile={isMobile}>
        <Header>Password Reset</Header>
        <SubHeader>
          <PlaceholderTitle>
            Enter your email and we will send you information on how to reset your password
          </PlaceholderTitle>
        </SubHeader>
        <Input
          title='Email'
          value={email}
          onChange={setEmail}
          styles={{ marginBottom: 16, marginTop: 60 }}
          placeholder='Enter your email...'
          inputStyles={isMobile ? { maxWidth: 320, minWidth: 200, width: '80vw' } : { minHeight: 48, minWidth: 320 }}
        />
        {error && (
          <ErrorWrapper isMobile={isMobile}>
            <ErrorText>{error}</ErrorText>
          </ErrorWrapper>
        )}
        {message && (
          <ErrorWrapper isMobile={isMobile}>
            <SuccessText>{message}</SuccessText>
          </ErrorWrapper>
        )}

        <Button
          disabled={loading || isEmpty(email)}
          type='submit'
          title='Reset password'
          handleButton={handleSubmit}
          styles={{
            marginTop: 24,
            marginBottom: 16,
            ...(isMobile ? { maxWidth: 320, minWidth: 200, width: '80vw' } : { width: 320 })
          }}
        />
        <LinkWrapper>
          <Link href='/sign-in'>Sign In</Link>
        </LinkWrapper>
        <LinkWrapper>
          Don&apos;t have an account? <Link href='/sign-up'>Sign Up</Link>
        </LinkWrapper>
      </Container>
    </Layout>
  );
};
