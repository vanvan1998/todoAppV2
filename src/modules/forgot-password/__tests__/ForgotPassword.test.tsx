import React from 'react';
import { render } from '@testing-library/react';
import { ForgotPassword } from '../ForgotPassword';

jest.mock('@next/third-parties/google', () => {
  return {
    GoogleAnalytics: () => <>GoogleAnalytics</>
  };
});

jest.mock('src/hooks', () => {
  const originalModule = jest.requireActual('src/hooks');
  return {
    ...originalModule,
    useMediaQuery: jest
      .fn()
      .mockReturnValue({
        isMobile: true
      })
      .mockReturnValueOnce({ isMobile: false })
  };
});

jest.mock('src/contexts', () => {
  return {
    useAuth: jest.fn().mockReturnValue({ currentUser: undefined, signIn: jest.fn() })
  };
});

jest.mock('next/navigation', () => {
  return {
    useRouter: jest.fn().mockReturnValue({
      push: jest.fn()
    })
  };
});

describe('ForgotPassword', () => {
  it('renders ForgotPassword with mobile view', () => {
    const component = render(<ForgotPassword />);
    expect(component).toMatchSnapshot();
  });

  it('renders ForgotPassword with desktop view', () => {
    const component = render(<ForgotPassword />);
    expect(component).toMatchSnapshot();
  });
});
