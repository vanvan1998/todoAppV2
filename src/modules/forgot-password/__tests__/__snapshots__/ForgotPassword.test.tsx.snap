// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword renders ForgotPassword with desktop view 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="width: 100%; display: flex; flex-direction: column; min-height: 100vh; background-color: rgb(239, 241, 249); align-items: center;"
      >
        <div
          class="sc-UblHX gPycWq"
        >
          <div
            class="sc-blHHSb kTKyqX"
          >
            Password Reset
          </div>
          <div
            class="sc-frniUE gDAuDV"
          >
            <div
              class="sc-egkSDF cGFidm"
            >
              Enter your email and we will send you information on how to reset your password
            </div>
          </div>
          <div
            style="margin-bottom: 16px; margin-top: 60px;"
          >
            <div
              class="sc-gtLWhw jPWJOW"
            >
              Email
            </div>
            <input
              aria-label="Email"
              class="sc-kLhKbu gjDzjv"
              placeholder="Enter your email..."
              style="max-width: 320px; min-width: 200px; width: 80vw;"
              type="text"
              value=""
            />
          </div>
          <button
            buttontype="primary"
            class="sc-ghWlax hYhlzt"
            disabled=""
            style="margin-top: 24px; margin-bottom: 16px; max-width: 320px; min-width: 200px; width: 80vw; background-color: rgb(182, 180, 185);"
            type="submit"
          >
            Reset password
          </button>
          <div
            class="sc-fufdwm dzrqEa"
          >
            <a
              href="/sign-in"
            >
              Sign In
            </a>
          </div>
          <div
            class="sc-fufdwm dzrqEa"
          >
            Don't have an account? 
            <a
              href="/sign-up"
            >
              Sign Up
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="width: 100%; display: flex; flex-direction: column; min-height: 100vh; background-color: rgb(239, 241, 249); align-items: center;"
    >
      <div
        class="sc-UblHX gPycWq"
      >
        <div
          class="sc-blHHSb kTKyqX"
        >
          Password Reset
        </div>
        <div
          class="sc-frniUE gDAuDV"
        >
          <div
            class="sc-egkSDF cGFidm"
          >
            Enter your email and we will send you information on how to reset your password
          </div>
        </div>
        <div
          style="margin-bottom: 16px; margin-top: 60px;"
        >
          <div
            class="sc-gtLWhw jPWJOW"
          >
            Email
          </div>
          <input
            aria-label="Email"
            class="sc-kLhKbu gjDzjv"
            placeholder="Enter your email..."
            style="max-width: 320px; min-width: 200px; width: 80vw;"
            type="text"
            value=""
          />
        </div>
        <button
          buttontype="primary"
          class="sc-ghWlax hYhlzt"
          disabled=""
          style="margin-top: 24px; margin-bottom: 16px; max-width: 320px; min-width: 200px; width: 80vw; background-color: rgb(182, 180, 185);"
          type="submit"
        >
          Reset password
        </button>
        <div
          class="sc-fufdwm dzrqEa"
        >
          <a
            href="/sign-in"
          >
            Sign In
          </a>
        </div>
        <div
          class="sc-fufdwm dzrqEa"
        >
          Don't have an account? 
          <a
            href="/sign-up"
          >
            Sign Up
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ForgotPassword renders ForgotPassword with mobile view 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="width: 100%; display: flex; flex-direction: column; min-height: 100vh; background-color: rgb(239, 241, 249); align-items: center;"
      >
        <div
          class="sc-UblHX gaOlys"
        >
          <div
            class="sc-blHHSb kTKyqX"
          >
            Password Reset
          </div>
          <div
            class="sc-frniUE gDAuDV"
          >
            <div
              class="sc-egkSDF cGFidm"
            >
              Enter your email and we will send you information on how to reset your password
            </div>
          </div>
          <div
            style="margin-bottom: 16px; margin-top: 60px;"
          >
            <div
              class="sc-gtLWhw jPWJOW"
            >
              Email
            </div>
            <input
              aria-label="Email"
              class="sc-kLhKbu gjDzjv"
              placeholder="Enter your email..."
              style="min-height: 48px; min-width: 320px;"
              type="text"
              value=""
            />
          </div>
          <button
            buttontype="primary"
            class="sc-ghWlax hYhlzt"
            disabled=""
            style="margin-top: 24px; margin-bottom: 16px; width: 320px; background-color: rgb(182, 180, 185);"
            type="submit"
          >
            Reset password
          </button>
          <div
            class="sc-fufdwm dzrqEa"
          >
            <a
              href="/sign-in"
            >
              Sign In
            </a>
          </div>
          <div
            class="sc-fufdwm dzrqEa"
          >
            Don't have an account? 
            <a
              href="/sign-up"
            >
              Sign Up
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="width: 100%; display: flex; flex-direction: column; min-height: 100vh; background-color: rgb(239, 241, 249); align-items: center;"
    >
      <div
        class="sc-UblHX gaOlys"
      >
        <div
          class="sc-blHHSb kTKyqX"
        >
          Password Reset
        </div>
        <div
          class="sc-frniUE gDAuDV"
        >
          <div
            class="sc-egkSDF cGFidm"
          >
            Enter your email and we will send you information on how to reset your password
          </div>
        </div>
        <div
          style="margin-bottom: 16px; margin-top: 60px;"
        >
          <div
            class="sc-gtLWhw jPWJOW"
          >
            Email
          </div>
          <input
            aria-label="Email"
            class="sc-kLhKbu gjDzjv"
            placeholder="Enter your email..."
            style="min-height: 48px; min-width: 320px;"
            type="text"
            value=""
          />
        </div>
        <button
          buttontype="primary"
          class="sc-ghWlax hYhlzt"
          disabled=""
          style="margin-top: 24px; margin-bottom: 16px; width: 320px; background-color: rgb(182, 180, 185);"
          type="submit"
        >
          Reset password
        </button>
        <div
          class="sc-fufdwm dzrqEa"
        >
          <a
            href="/sign-in"
          >
            Sign In
          </a>
        </div>
        <div
          class="sc-fufdwm dzrqEa"
        >
          Don't have an account? 
          <a
            href="/sign-up"
          >
            Sign Up
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
